kibana-reports/server/routes/utils/visual_report/style.css

```diff
@@ -0,0 +1,211 @@
+#reportingHeader h1:first-child,
+#reportingFooter h1:first-child,
+#reportingHeader h2:first-child,
+#reportingFooter h2:first-child,
+#reportingHeader h3:first-child,
+#reportingFooter h3:first-child,
+#reportingHeader h4:first-child,
+#reportingFooter h4:first-child,
+#reportingHeader h5:first-child,
+#reportingFooter h5:first-child,
+#reportingHeader h6:first-child,
+#reportingFooter h6:first-child {
+  margin-top: 0.25em;
+}
+
+/* nicer list styles */
+#reportingHeader ul,
+#reportingFooter ul,
+#reportingHeader ol,
+#reportingFooter ol {
+  max-width: 70rem;
+  margin-bottom: 1em;
+}
+
+#reportingHeader ul li,
+#reportingFooter ul li,
+#reportingHeader ol li,
+#reportingFooter ol li {
+  margin-bottom: 0.25em;
+  margin-left: -0.5em;
+  padding-left: 0.25em;
+}
+
+#reportingHeader ul,
+#reportingFooter ul {
+  list-style-type: disc;
+}
+
+/* here we explicitly set nested paragraphs inside lists to inherit their styles from the list, in case markdown does funky things */
+#reportingHeader ul p,
+#reportingFooter ul p,
+#reportingHeader ol p,
+#reportingFooter ol p {
+  font-family: inherit;
+  font-size: inherit;
+  font-weight: inherit;
+  /* We only inherit vertical spacing, not horizontal */
+  margin-top: inherit;
+  margin-bottom: inherit;
+}
+</style>
+  <style>
+    /* mde style sheet*/
+  .mde-preview .mde-preview-content {
+    padding: 10px;
+  }
+.mde-preview .mde-preview-content p,
+.mde-preview .mde-preview-content blockquote,
+.mde-preview .mde-preview-content ul,
+.mde-preview .mde-preview-content ol,
+.mde-preview .mde-preview-content dl,
+.mde-preview .mde-preview-content table,
+.mde-preview .mde-preview-content pre {
+  margin-top: 0;
+  margin-bottom: 16px;
+}
+.mde-preview .mde-preview-content h1,
+.mde-preview .mde-preview-content h2,
+.mde-preview .mde-preview-content h3 {
+  margin-top: 24px;
+  margin-bottom: 16px;
+  font-weight: 600;
+  line-height: 1.25;
+  padding-bottom: 0.3em;
+}
+.mde-preview .mde-preview-content h1 {
+  font-size: 1.6em;
+  border-bottom: 1px solid #eee;
+}
+.mde-preview .mde-preview-content h2 {
+  font-size: 1.4em;
+}
+.mde-preview .mde-preview-content h3 {
+  font-size: 1.2em;
+}
+.mde-preview .mde-preview-content ul,
+.mde-preview .mde-preview-content ol {
+  padding-left: 2em;
+}
+.mde-preview .mde-preview-content blockquote {
+  margin-left: 0;
+  padding: 0 1em;
+  color: #777;
+  border-left: 0.25em solid #ddd;
+}
+.mde-preview .mde-preview-content blockquote > :first-child {
+  margin-top: 0;
+}
+.mde-preview .mde-preview-content blockquote > :last-child {
+  margin-bottom: 0;
+}
+.mde-preview .mde-preview-content code {
+  padding: 0.2em 0 0.2em 0;
+  margin: 0;
+  font-size: 90%;
+  background-color: rgba(0, 0, 0, 0.04);
+  border-radius: 3px;
+}
+
+.mde-preview .mde-preview-content pre {
+  padding: 16px;
+  overflow: auto;
+  font-size: 85%;
+  line-height: 1.45;
+  background-color: #f7f7f7;
+  border-radius: 3px;
+}
+.mde-preview .mde-preview-content pre code {
```

Comment:
just curious are these styles taken from react-mde library? if so we might be able to include them instead of defining them again

Reply:
I believe so. But how can we import it from the library?

Reply:
it should be `@import 'react-mde/lib/styles/css/react-mde-preview.css';` or `@import '~react-mde/lib/styles/css/react-mde-preview.css';`, but i tried both didn't work. Not sure if kibana or pdf generation did something special, maybe it's easier to import the css in typescript in `addReportStyle` and merge it with `visual_report/style.css`, i can look into it later

Reply:
it works by just changing the [css variable](https://github.com/zhongnansu/kibana-reports/blob/ebdfe99556966d11a28e03e9f497130980a293bd/kibana-reports/server/routes/utils/visual_report/visualReportHelper.ts#L195-L196) to 
```js
const css = fs.readFileSync(`${__dirname}/style.css`).toString() + fs.readFileSync(`${__dirname}/../../../../node_modules/react-mde/lib/styles/css/react-mde-preview.css`).toString();
```

but seems the current way might be better.. since we don't rely on css path in the library

Reply:
Thanks for all the investigation. I do agree that rely on css path in lib is kind of tricky. So let's keep it the existing way till we have a better solution